client/router.js

```diff
@@ -268,12 +277,9 @@ Router.prototype._notfoundRoute = function(context) {
 Router.prototype.initialize = function() {
   var self = this;
 
-  this._middleware.push(function (ctx, next) {
-    setTimeout(function() {
-      var str = location.search.slice(1);
-      ctx.params.query = self._qs.parse(str);
-      next();
-    }, 0);
+  this._triggersEnter.push(function(ctx) {
+    var str = location.search.slice(1);
```

Comment:
I think here we can't depend on the `location`. I think right now, location is not yet changed?
Quite not sure, I'll check.